src/sqlfluff/core/parser/segments/base.py

```diff
@@ -661,6 +661,25 @@ def get_children(self, *seg_type):
                 buff.append(seg)
         return buff
 
+    def select_children(self, start_seg: 'BaseSegment' = None,
```

Comment:
I think this function, or something similar to it, may simplify some of the typical logic needed when generating fixes. Thoughts?